docs/querying/dart.md

```diff
@@ -0,0 +1,140 @@
+
+### API
+
+Dart uses the SQL endpoint `/druid/v2/sql`. To use Dart, include the query context parameter `engine` and set it to `msq-dart`:
+
+<Tabs>
+<TabItem value="SET" label="SET" default>
+
+As part of your query using `SET engine = 'msq-dart'`:
+
+```json
+"query":"SET \"engine\"='msq-dart';\nSELECT\n  user,\n  commentLength,\n  COUNT(*) AS \"COUNT\"\nFROM \"wikipedia\"\nGROUP BY 1, 2\nORDER BY 2 DESC"
+```
+
+</TabItem>
+
+<TabItem value="context_block" label="Context block">
+
+As part of a `context` block: 
+
+```json
+{
+  "query": "SELECT\n  user,\n  commentLength,\n  COUNT(*) AS \"COUNT\"\nFROM \"wikipedia\"\nGROUP BY 1, 2\nORDER BY 2 DESC",
+  "context": {
+    "engine": "msq-dart"
+  }
+}
+```
+
+
+  </TabItem>
+  </Tabs>
+
+## Query context parameters
+
+You can use query context parameters to control Dart's behavior. The following table lists the supported query context parameters:
+
+| Parameter | Description | Default value |
+|---|---|---|
+| `finalizeAggregations` | Determines the type of aggregation to return. If true, Druid finalizes the results of complex aggregations that directly appear in query results. If false, Druid returns the aggregation's intermediate type rather than finalized type. This parameter is useful during ingestion, where it enables storing sketches directly in Druid tables. For more information about aggregations, see [SQL aggregation functions](../querying/sql-aggregations.md). | `true` |
+| `includeSegmentSource` |  Controls the sources that are queried for results in addition to the segments present on deep storage. Can be `NONE` or `REALTIME`. If this value is `NONE`, only non-realtime (published and used) segments will be downloaded from deep storage. If this value is `REALTIME`, results will also be included from realtime tasks.|  `REALTIME` |
+| `removeNullBytes` |The MSQ engine cannot process null bytes in strings and throws `InvalidNullByteFault` if it encounters them in the source data. If the parameter is set to true, The MSQ engine will remove the null bytes in string fields when reading the data. | `false` |
+|`maxConcurrentStages`|Number of stages that can run concurrently for a query. A higher number can potentially improve pipelining but results in less memory available for each stage.|2|
+|`maxNonLeafWorkers`|Number of workers to use for stages beyond the leaf stage| 1 (Scatter-gather style)|
+| `sqlJoinAlgorithm` | Algorithm to use for JOIN. Use `broadcast` (the default) for broadcast hash join or `sortMerge` for sort-merge join. Affects all JOIN operations in the query. This is a hint to the MSQ engine and the actual joins in the query may proceed in a different way than specified. See [Joins](#joins) for more details. | `broadcast` |
+|`targetPartitionsPerWorker`|Number of partitions Druid generates for each worker. This number controls how much parallelism can be maintained throughout a query.|1|
+
+
+  ## Known issues and limitations
+
+- Dart doesn't do the following:
```

Comment:
please also mention that `useApproximateTopN` is not implemented; Dart always does exact topNs